### PR TITLE
v2 migration: `guides/development` to next

### PR DESCRIPTION
**Pull request by vasfvitor**
#### What kind of changes does this PR include?

- New or updated content
- Changes to the docs site code

#### Description

Pages: 
https://tauri.app/v1/guides/development/development-cycle
https://tauri.app/v1/guides/development/updating-dependencies

- Added redirects
- Diff at https://github.com/tauri-apps/tauri-docs/commit/3b7adc3ba9da1fc80e6e3f98575bb2958692b9b0
- Sidebar updated according to https://starlight.astro.build/guides/sidebar/#autogenerated-groups
- Modified the card below from /development-cycle: bold text, caution box and code filenames/path:

v2 (next):

Original from v1:

sidebar preview:

#### Personal Notes (can be ignored)

`import Tabs from '@theme/Tabs';`
`import TabItem from '@theme/TabItem';`

can be replaced by `import { Tabs, TabItem } from '@astrojs/starlight/components';` to place code, if needed
or  `import CommandTabs from '@components/CommandTabs.astro';` to place commands

This `import Command from '@theme/Command'`
to `import CommandTabs from '@components/CommandTabs.astro';`

The Command
`<Command name="dev" />
`
becomes
```
<CommandTabs
	npm="npm run tauri dev"
	yarn="yarn tauri dev"
	pnpm="pnpm tauri dev"
	cargo="cargo tauri dev"
/>
```

https://starlight.astro.build/guides/authoring-content/#note-aside
`
:::info title`
into 
`:::caution[title]`

`:::info` can be: `note`, `tip`, `caution` or `danger` depending on context.

### CODE REVIEW

File: src/content/docs/fr/guides/develop/development-cycle.mdx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import CommandTabs from '@components/CommandTabs.astro';
+
+### 1. Démarrez votre serveur de développement
+
+Maintenant que tout est configuré, vous devez démarrer votre serveur de développement d'applications fourni par votre framework ou bundler d'interface utilisateur (en supposant que vous en utilisiez un, bien sûr).
+
+:::note
+
+Chaque framework possède ses propres outils de développement. Il n'entre pas dans le cadre de ce document de les couvrir tous ou de rester à jour.
+:::
```
Comment (lorenzolewis) [inline src/content/docs/fr/guides/develop/development-cycle.mdx]:
```suggestion
:::note

Chaque framework possède ses propres outils de développement. Il n'entre pas dans le cadre de ce document de les couvrir tous ou de rester à jour.

:::
```

File: src/content/docs/guides/develop.mdx
Comment (lorenzolewis) [inline src/content/docs/guides/develop.mdx:1]: I'd like to keep this page as a sort of "jumping off" page for the respective sub-pages under the Develop section.

File: src/content/docs/guides/develop/development-cycle.mdx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import CommandTabs from '@components/CommandTabs.astro';
+
+### 1. Start Your Dev server
+
+Now that you have everything set up, you should start your application development server provided by your UI framework or bundler (assuming you're using one, of course).
+
+:::note
+
+Every framework has its own development tooling. It is outside of the scope of this document to cover them all or stay up to date.
+:::
```
Comment (lorenzolewis) [inline src/content/docs/guides/develop/development-cycle.mdx]:
```suggestion
:::note

Every framework has its own development tooling. It is outside of the scope of this document to cover them all or stay up to date.

:::
```

File: src/content/docs/zh-CN/guides/develop/development-cycle.mdx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import CommandTabs from '@components/CommandTabs.astro';
+
+### 1. 启动开发服务器
+
+现在您已经设置好了一切，您应该启动由 UI 框架或捆绑器提供的应用程序开发服务器（当然，假设您使用的是）。
+
+:::note
+
+每个框架都有自己的开发工具。 涵盖所有内容或保持最新状态超出了本文档的范围。
+:::
```
Comment (lorenzolewis) [inline src/content/docs/zh-CN/guides/develop/development-cycle.mdx]:
```suggestion
:::note

每个框架都有自己的开发工具。 涵盖所有内容或保持最新状态超出了本文档的范围。

:::
```

**Review by lorenzolewis — CHANGES_REQUESTED** [review]
Thank you! @vasfvitor ! A few notes:

- KO and IT translations: I saw your message on Discord this morning (just getting back in). I think we should transfer those translations into this branch as well but keep them deactivated. I haven't looked into the best way to do this (whether via drafts, using an underscore prefix, etc.) but I saw you had a local draft with that so will take your guidance on that
- Instead of having all of the sub-pages in the sidebar I'd prefer to just keep the "parent" page, so in this case "Develop". Then on that landing page you could use the `LinkCard` component to jump out to those. One UX item I don't like with that is the parent's sidebar entry won't be highlighted when on a sub-page. I've put in a discussion for this in https://github.com/withastro/starlight/discussions/950 and I know that Astro's main docs are also wanting similar behaviour. We can always do a component override to get this functionality in the meantime.

File: src/content/docs/guides/develop.mdx
Comment (vasfvitor) [reply src/content/docs/guides/develop.mdx:1]: Ok, I'll undelete it. Also I'll keep an eye on those lines in mdx.

**Comment by vasfvitor** [comment]
About the translations: Ok, I'll look what I've done here

About the highlight: I've been following the Astro Docs migration to Starlight and they at least solved this in their code as you can see here https://docs-git-fork-theotterlord-starlight-migration-astrodotbuild.vercel.app/en/tutorial/1-setup/
So I consider every parent page will have this `LinkCard` component.
I'll go through each PR this week

**Comment by vasfvitor** [comment]
Don't know if this is trivial but I guess it's up to you. Consider that having descriptions we'd have to write each one, copy from pages contents (as we don't have yet unique descriptions for each page) . With this in mind the simplest way would be 3. List without description. It's already committed this way (3), but in case you want to change, please.

TODO: Highlight the sidebar. I guess a dedicated PR for this one is better.

Added a commit where the `develop.mdx` page was moved to `develop/index.mdx`. The end `/` is not necessary in Astro Config for it to work. If this is ok, then we can settle as the default way to migrate. Same as the `Link Card` layout, I'd say.

Italian i18n have the /development pages but they are not translated

1. Grid with description:



<br/><br/>
2. Grid without description:



<br/><br/>
3. List without description


<br/><br/>
4. List with description

**Comment by lorenzolewis** [comment]
Awesome! And yes, I heard Sarah say at one point that Astro docs wanted that sub-item highlighting, but didn't take the time to see if/how they'd implemented it. So awesome to see!

I prefer option 1 with the grid with descriptions. One approach you could use to reduce duplicating those descriptions would be to pull in the content collection entry and get the respective title and description (if available) from there. I've done a similar thing in this component: https://github.com/tauri-apps/tauri-docs/blob/next/src/components/list/Features.astro (note: I haven't tested if it works correctly for i18n'd pages, I'm 99% sure it does not).

I actually have been wanting to build on top of `LinkCard` to allow passing a slug to it and then have the component itself automatically pull in the respective i18n's title, description, and link and PR that upstream, but haven't had the chance. 

Would you be interested in wrapping `LinkCard` to give that functionality? Then that can be easily applied to the rest of the migrations as well.

**Comment by vasfvitor** [comment]
As of now, all pages shares the same description from v1 docs. The ones in the image was copied from page contents as an example. And from what I've seen in v1 pages, it's the same deal: shared description from the `_category_.yml` file.

Just to be sure we'll settle with 1, then fallback to 2? Will we write copy for each page?

I'm not sure I follow you on the "PR that upstream" part. I understood: create a custom component and open a separated PR here for it to be available.

To update, I'm already playing with a new/replacement component (based on `Features.astro`) that works with i18n. 

Aiming to make available translated content and fallback in the list, and if I manage, add a "fallback" notice on the card.

So far it kinda works for both i18n translated and default pages, but not for fallback, yet. Almost there

**Comment by lorenzolewis** [comment]
Sorry, I could have been a little more clear in my explanation, so my bad.

Let's go with how you currently have it setup for this PR (with just the title and link) and then we can work through #1717 in parallel then update once that's done (if needed). I'll put a few more comments on that PR directly for that.

Does that all sound alright? Ready to merge this in?

**Comment by vasfvitor** [comment]
updated to be list, for now without description.
created index page for each i18n page
now it's ready

**Comment by lorenzolewis** [comment]
Perfection! Thank you very much! Hopefully the other will be a bit easier now to push through.

**Review by lorenzolewis — APPROVED** [review]

**Pull request merged**